### PR TITLE
Dev dax 2 - support for pmempool and tests

### PR DESCRIPTION
**Pull request by wojtuss**

### CODE REVIEW

File: src/common/file.c [hunk 1]
```diff
@@ -313,6 +312,12 @@ util_file_create(const char *path, size_t size, size_t minsize)
 
 	ASSERTne(size, 0);
 
+	if (util_file_is_device_dax(path)) {
```
Comment (pbalcer) [inline src/common/file.c]: remove this if (I'm assuming rebase issue)

File: src/common/mmap.h [hunk 2]
```diff
@@ -82,6 +82,19 @@ int util_range_none(void *addr, size_t len);
 char *util_map_hint_unused(void *addr, size_t len, size_t align);
 char *util_map_hint(size_t len, size_t req_align);
 
+#ifdef DEBUG
```
Comment (pbalcer) [inline src/common/mmap.h]: remove those macros.

File: src/common/file.c (hunk 1 above)
Comment (wojtuss) [reply src/common/file.c]: done

File: src/common/mmap.h (hunk 2 above)
Comment (wojtuss) [reply src/common/mmap.h]: done

**Pull request merged**